lib/acme/client/resources/order.rb

```diff
@@ -26,12 +27,20 @@ def finalize(csr:)
 
   def certificate
     if certificate_url
-      @client.certificate(url: certificate_url)
+      certificate, @alternate_links = @client.certificate(url: certificate_url)
+      certificate
     else
       raise Acme::Client::Error::CertificateNotReady, 'No certificate_url to collect the order'
     end
   end
 
+  def alternative_certificates
+    @alternate_links.map do |certificate_url|
```

Comment:
one more bit of feedback here. If the `certificate` method hasn't been invoked yet, this method will return an empty array. I think the default value of the `@alternate_links` instance variable should be `nil` so that we can invoke the `certificate` method ourselves when it hasn't been called yet.

```suggestion
  def alternative_certificates
    _ = certificate if @alternate_links.nil?
    @alternate_links.map do |certificate_url|
```

Reply:
Without this change, the caller must know to call the `certificate` method first, which isn't an ideal API in my opinon.

Reply:
I agree. The design of the client is to hide the underlying http call from the user.